auto3dseg/algorithm_templates/dints/scripts/algo.py

```diff
@@ -24,13 +25,11 @@
 logger = get_logger(module_name=__name__)
 
 
-def get_gpu_available_memory():
-    command = "nvidia-smi --query-gpu=memory.free --format=csv"
-    memory_free_info = (
-        subprocess.check_output(command.split()).decode("ascii").split("\n")[:-1][1:]
-    )
-    memory_free_values = [int(x.split()[0]) for i, x in enumerate(memory_free_info)]
-    return memory_free_values
+def get_mem_from_visible_gpus():
+    available_mem_visible_gpus = []
+    for d in range(torch.cuda.device_count()):
+        available_mem_visible_gpus.append(torch.cuda.mem_get_info(device=d)[0])
+    return available_mem_visible_gpus
```

Comment:
Hi @myron thank you for the suggestion. Here are some things I found:

- I also tried `torch.cuda.mem_get_info(device=d)` and found out that it would show the total memory, instead of the free memory. For example, if GPU memory is 12GB, and 1GB is used for display, or used by another program, `torch.cuda.mem_get_info(device=d)` will give `(11, 12)` \*, but  `torch.cuda.get_device_properties(i).total_memory` will give `12`*. 
- As I test it, the call of `torch.cuda.mem_get_info(device=d)` would consume about 250MB GPU memory on the device `d`, which seems quite significant.
- I did not find other ways to find pythonic way to query GPU memory via `cuda`.

The original logic was to find the free memory. Does it matter if we change it to total? If it does, would the use of 250MB worth it? @dongyang0122 

\*unit should be bytes, but I'm using gigabytes just to give an idea.

Reply:
It is better to compute available GPU memories instead of total ones. I feel the way to compute available GPU memories in the PR is okay. Although the additional 200~300 MB could be a potential concern, there are many alternatives to serve the same purpose except calling "nvidia-smi" library.

Reply:
Thanks @dongyang0122 . I will make no changes based on your comments then.